oak_sev_guest/src/ghcb.rs

```diff
@@ -318,6 +328,56 @@ where
         )?)
     }
 
+    /// Read a 32-bit value from a MMIO memory address via the MMIO Access protocol.
+    ///
+    /// See Section 4.1.5 in <https://www.amd.com/content/dam/amd/en/documents/epyc-technical-docs/specifications/56421-guest-hypervisor-communication-block-standardization.pdf>.
+    pub fn mmio_read_u32(&mut self, source: PhysAddr) -> Result<u32, &'static str> {
+        let gpa_base = self.get_gpa().as_u64();
+        let ghcb = self.ghcb.as_mut();
+
+        ghcb.reset();
+        ghcb.sw_exit_code = SW_EXIT_CODE_MMIO_READ;
+        ghcb.sw_exit_info_1 = source.as_u64();
+        ghcb.sw_exit_info_2 = core::mem::size_of::<u32>() as u64;
+        // Use the shared_buffer as the unencrypted guest memory. This is mandated as of Version
+        // 2 of the communication block.
+        ghcb.sw_scratch = gpa_base + 0x800;
```

Comment:
isn't there something in `core::mem` that can calculate the offset of the field in the struct?